six/modules/c++/six/include/six/ReadControl.h

```diff
@@ -108,12 +109,43 @@ class ReadControl
      *  This function reads in the image area specified by the region.
      *  If you want us to use a work buffer, you can set it through region.
      *
+     *  To simply read the entire image, pass a default-constructed
+     *  Region as the first parameter
+     *
      *  Once read, the image buffer is set in both the region pointer,
      *  and in the return value, for convenience
+     *
+     *  For safety, prefer the overload below.
      */
     virtual UByte* interleaved(Region& region, size_t imageNumber) = 0;
 
     /*!
+     *  This function reads in the image area specified by the region.
+     *  If you want us to use a work buffer, you can set it through region.
+     *
+     *  To simply read the entire image, pass a default-constructed
```

Comment:
Also add a comment somewhere in here that there's no casting that happens... for example, if it's an integer SICD on disk and you pass in a `ScopedArray<std::complex<float>>`, we're not going to cast the data to complex float for you; it'll just do the wrong thing.